templates/backup-pre-mysql.sh

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+
+# Exits after first failure
+set -e
+
+TMPDIR="{{ MYSQL_SERVER_BACKUP_DIR }}"
+timestamp=$(date +%Y%m%dT%H:%M:%S)
+
+rm -rf "${TMPDIR}"
+mkdir -p "${TMPDIR}"
+mysql -s -e 'show databases' | tail -n +1 | grep -v performance_schema | grep -v information_schema | while read DBNAME; do mysqldump -uroot --complete-insert --events --single-transaction "${DBNAME}" > "${TMPDIR}/${DBNAME}".sql; done
```

Comment:
So `mysql -s e show databases` output something like this: 
mysql -s -e 'show databases' -u root -p

```
Database
information_schema
amarokdb
dalite_ng
mysql
oscar_travis
performance_schema
```
tail -n +1 outputs everything but the first line 
grep -v removes two magical schemas that can't be dumped.